src/components/ZoomableImage.tsx

```diff
@@ -72,41 +81,88 @@ const ZoomableImage = ({
         open={isOpen}
         onClose={() => {
           setIsOpen(false);
+          setUseFullResolution(false);
+          setLoadingFullResolution(false);
         }}
         maxWidth="xl"
         fullScreen={fullScreen}
       >
-        <Box
-          sx={{
-            p: 1,
-            display: "flex",
-            flexDirection: "row-reverse",
-          }}
-        >
-          <IconButton onClick={() => setIsOpen(false)}>
-            <CloseIcon />
-          </IconButton>
-        </Box>
+        {loadingFullResolution && (
+          <Box
+            sx={{
+              position: "absolute",
+              inset: 0,
+              display: "flex",
+              justifyContent: "center",
+              alignItems: "center",
+              backgroundColor: "rgba(0, 0, 0, 0.6)",
+              zIndex: 10,
+            }}
+          >
+            <IconButton onClick={() => setIsOpen(false)}>
+              <CloseIcon />
+            </IconButton>
+            <Loader />
+          </Box>
+        )}
         <Divider />
         <DialogContent
           sx={{
             p: { xs: 1, md: 2 },
+            position: "relative",
           }}
         >
           <TransformWrapper limitToBounds={false} ref={apiRef}>
             <TransformComponent>
               <img
-                src={srcFull ?? src}
+                src={useFullResolution && srcFull ? srcFull : src}
                 alt=""
+                loading="lazy"
                 style={{
-                  maxHeight: "calc(100vh - 160px )",
-                  maxWidth: "min(100%, 1400px)",
+                  maxHeight: "calc(100vh - 160px)",
+                  maxWidth: "100%",
+                  width: "100%",
+                  height: "auto",
+                  objectFit: "contain",
                   transform: `rotate(${rotation * 90}deg)`,
                   transformOrigin: "center",
+                  display: "block",
+                  margin: "auto",
                 }}
               />
             </TransformComponent>
           </TransformWrapper>
+          {srcFull && (
+            <Button
+              disabled={loadingFullResolution}
+              variant={useFullResolution ? "contained" : "outlined"}
```

Comment:
outlined for all looks better. Otherwise the button can disapear because of transparency

<img width="180" height="70" alt="image" src="https://github.com/user-attachments/assets/1cdf435f-193f-4557-80d4-f84bde68b1de" />